File: linda-presentation/src/main/java/it/unibo/coordination/linda/presentation/StringTupleSerializer.java
package it.unibo.coordination.linda.presentation;

import com.fasterxml.jackson.databind.ObjectMapper;
import it.unibo.coordination.linda.text.StringTuple;

import static it.unibo.coordination.utils.CollectionUtils.mapOf;

class StringTupleSerializer extends DynamicSerializer<StringTuple> {

    public StringTupleSerializer(MIMETypes mimeType, ObjectMapper mapper) {
        super(mimeType, mapper);
    }

    @Override
    public Object toDynamicObject(StringTuple object) {
        return mapOf("tuple", object.getValue());
    }
}

File: linda-text-presentation/src/main/kotlin/it/unibo/coordination/linda/text/StringTupleSerializer.kt
package it.unibo.coordination.linda.text

import com.fasterxml.jackson.databind.ObjectMapper
import it.unibo.presentation.DynamicSerializer
import it.unibo.presentation.MIMETypes

internal class StringTupleSerializer(mimeType: MIMETypes, mapper: ObjectMapper) : DynamicSerializer<StringTuple>(mimeType, mapper) {
    override fun toDynamicObject(`object`: StringTuple): Any {
        return mapOf("tuple" to `object`.value)
    }
}